src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -52,29 +52,44 @@ public SonarQubeClient(HttpMessageHandler messageHandler, TimeSpan requestTimeou
                 throw new ArgumentException("Doesn't expect a zero or negative timeout.", nameof(requestTimeout));
             }
 
-            this.messageHandler = messageHandler;
-            this.requestTimeout = requestTimeout;
+            this.httpClient = new HttpClient(messageHandler)
+            {
+                BaseAddress = connection.ServerUri,
+                Timeout = requestTimeout
+            };
+            this.httpClient.DefaultRequestHeaders.Authorization = AuthenticationHeaderFactory.Create(connection);
+        }
+
+        public void Dispose()
+        {
+            if (isDisposed)
+            {
+                return;
+            }
+
+            this.httpClient.Dispose();
```

Comment:
Is this going to dispose the supplied _messageHandler_ too?

Reply:
Yes! Default behavior.